services/src/main/java/uk/gov/moj/sdt/services/config/ServicesConfig.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.moj.sdt.services.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.jms.listener.DefaultMessageListenerContainer;
+import org.springframework.jms.listener.adapter.MessageListenerAdapter;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import uk.gov.moj.sdt.services.messaging.QueueConfig;
+import uk.gov.moj.sdt.services.utils.GenericXmlParser;
+import uk.gov.moj.sdt.services.utils.IndividualRequestsXmlParser;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.jms.ConnectionFactory;
+
+@ComponentScan("uk.gov.moj.sdt")
+@Configuration
+@EnableConfigurationProperties(QueueConfig.class)
+@EnableTransactionManagement
+public class ServicesConfig {
+
+    @Autowired
+    private ConnectionFactory jmsConnectionFactory;
+
+    @Autowired
+    private PlatformTransactionManager transactionManager;
+
+    @Autowired
+    @Lazy
+    private MessageListenerAdapter messageListenerAdapter;
+
+    @Autowired
+    private QueueConfig queueConfig;
+
+    @Bean
+    @Lazy
+    @Qualifier("messageListenerContainer")
+    public DefaultMessageListenerContainer messageListenerContainer() {
+        DefaultMessageListenerContainer defaultMessageListenerContainer = new DefaultMessageListenerContainer();
+        defaultMessageListenerContainer.setConnectionFactory(jmsConnectionFactory);
+        defaultMessageListenerContainer.setDestinationName(queueConfig.getQueueConfig().get("MCOL"));
+        defaultMessageListenerContainer.setMessageListener(messageListenerAdapter);
+        defaultMessageListenerContainer.setTransactionManager(transactionManager);
+        defaultMessageListenerContainer.setConcurrentConsumers(1);
+        defaultMessageListenerContainer.setMaxConcurrentConsumers(5);
+        defaultMessageListenerContainer.setReceiveTimeout(30000);
+        defaultMessageListenerContainer.setIdleTaskExecutionLimit(10);
+        defaultMessageListenerContainer.setIdleConsumerLimit(5);
```

Comment:
Should the hard coded values be replaced with ones read from an external source such as a properties file?

Reply:
created entries in application.yml file